### ISSUE TITLE
Some angles with low snappiness not getting enough of a bonus

### ISSUE THREAD

**Issue by MBmasher** [Syriiin/UniCircleDifficulty#4]
90 degree angles with -1 snappiness give only a 0.5 bonus which seems rather low for their actual difficulty.

An idea is to use a logarithmic curve instead of a sine curve for objects with low snappiness.

**Comment by Syriiin** [comment]
My current task is actually a rewrite of the angle system to make it both less exploitable and more rewarding for harder patterns. Should be seeing the change come around in the next couple weeks but can't say for sure.

**Comment by Syriiin** [comment]
With the changes in 10945b27266914d3d70126bbc4b1ea1b20c9249e, there is no longer an "angle system" as such, and instead models skills such as aim as "dimensional skills" where the actions described by the difficulty points are vectors of the motion required for the action (jump). These actions are completed in 1 of 2 ways: snapping or flowing.

There are 2 differences between them in regards to how they are calculated:

1. Snapping has 2 forces applied per action - acceleration, and deceleration - whereas flowing only has the consistent movement force.
This means snapping uses more energy.
2. Snapping has a different position/time curve than flowing. Flowing is simply a straight line since the speed is constant (no accel or decel). But snapping, since it has accel and decel forces, has a curve that looks something like this [this red line](https://i.imgur.com/JuPZIAY.png). 
This means that snapping has your cursor within the target area (circle radius) for a longer period of time than flowing.
The time within the target area is taken as the raw difficulty of an action (raw difficulty is the difficulty of the action without considering context).

In this new model, snappiness is determined by the volatility of the force fluctuations. ie, many different aim points in a small amount of time (like streams) means high volatility and thus is considered low snappiness.

Due the the way difficulty is measured for these type of skills at the moment, the extra difficulty of wide angle jumps isn't part of the raw difficulty, but instead is part of the energy exerted, and contributes more to the exertion. This results in a larger bonus over time, for example, square practice maps would get a sizable bonus to exertion.

This may change however as I am considering adding a multiplier to the raw difficulty based on the difference in vector direction.

TL;DR: aiming works completely differently now and in theory should reward harder patterns and streams much better than before.

**Comment by Syriiin** [comment]
After some thought I have decided against the idea of adding a volatility multiplier to raw difficulty as it would go against the purpose of it being the difficulty of the action in a vacuum.

Instead perhaps a better idea, if we desire to give force fluctuations literal impact in dimensional skill difficulty, would be to simply add the volatility as a difficulty factor much like exertion is for physical skills.
This will require some more thought however, as the current volatility value disregards flow since its purpose is to define the switch to flow.
If this idea is to be implemented, either the current value would need to be modified to increase with regard to the current snappiness (defined by a function of the volatility value itself) as well as balance it according to this change, or alternatively add a separate value.
